lib/collection/src/recommendations.rs

```diff
@@ -56,22 +56,28 @@ fn avg_vectors<'a>(
                 sparse_count += 1;
                 avg_sparse = vector.combine_aggregate(&avg_sparse, |v1, v2| v1 + v2);
             }
-            VectorRef::MultiDense(vector) => {
-                multi_count += 1;
-                avg_multi = Some(avg_multi.map_or_else(
-                    || vector.to_owned(),
-                    |mut avg_multi| {
-                        avg_multi
-                            .flattened_vectors
-                            .extend_from_slice(vector.flattened_vectors);
-                        avg_multi
-                    },
-                ));
+            VectorRef::MultiDense(multi_vector) => {
+                for vector in multi_vector.multi_vectors() {
+                    in_multi_count += 1;
+                    avg_multi = Some(avg_multi.map_or_else(
+                        || vector.to_owned(),
+                        |mut avg_multi| {
+                            for i in 0..vector.len() {
+                                if i >= avg_multi.len() {
+                                    avg_multi.push(vector[i])
+                                } else {
+                                    avg_multi[i] += vector[i];
+                                }
```

Comment:
the second dimensionality in multivectors is configured on a collection level, it should not be possible to get into else branch